### ISSUE TITLE
🤗 Retrieving a CSRF value with CSRF middleware

### ISSUE THREAD

**Issue by anthonycook** [gofiber/fiber#1193]
I can't seem to find any obvious method in the docs for retrieving a CSRF token value, I want to pass the value to my templates so it can be included in my HTML forms.

### Fiber route
```go
func RenderSignupPage(c *fiber.Ctx) error {

	return c.Render("signup", fiber.Map{
		"csrf": MY_TOKEN_VALUE
	})

}
```

### HTML form
```html
  <form method="POST" action="/signup">
    <input type="hidden" name="csrf" value="{{ csrf }}">
    ....
    <button type="submit">Submit</button>
  </form>
```

**Comment by bannedev** [comment]
Well there are two " hacky " ways through which you can do this.

- Define your own configs and use a custom key generator ( as in [here](https://github.com/gofiber/fiber/tree/master/middleware/csrf#custom-config))

- Use `c.Response().Header.Peek("Set-Cookie")` to get the Header and parse it accordingly.

Consider the fact that I just started working with Fiber too in this 🙈

**Comment by crossphoton** [comment]
Well there are two " hacky " ways through which you can do this.

- Define your own configs and use a custom key generator ( as in [here](https://github.com/gofiber/fiber/tree/master/middleware/csrf#custom-config))

- Use `c.Response().Header.Peek("Set-Cookie")` to get the Header and parse it accordingly.

Consider the fact that I just started working with Fiber too in this 🙈

**Comment by hi019** [comment]
A GET request should associate a CSRF token with the request. Does that fit your usecase? Ex `GET /signup`

**Comment by anthonycook** [comment]
> A GET request to any route should associate a CSRF token with the request. Does that fit your use-case? Ex `GET /signup`
> 
> I agree though, we should have a method to generate or get the CSRF token for a request. Laravel has an [identical feature](https://laravel.com/docs/8.x/csrf#preventing-csrf-requests). I'll make a PR.

I do make a GET request to /signup which renders my signup page and sets the CSRF token in a cookie, but my understanding is that the CSRF token needs to be submitted with my signup HTML form for the CSRF check to pass (double submit method).

Similar to your Laravel example, in packages like [github.com/gorilla/csrf](https://github.com/gorilla/csrf) you'd just pass the csrf.Token method to your HTML template which returns the token associated with that request.

Wasn't sure how Fiber users have got around this until I seen @crossphoton reply, which as he mentions aren't the most elegant solutions.

Not sure if I'm missing something or if this packages approach to CSRF works differently in terms to submitting HTML forms?

**Comment by hi019** [comment]
> I do make a GET request to /signup which renders my signup page and sets the CSRF token in a cookie, but my understanding is that the CSRF token needs to be submitted with my signup HTML form for the CSRF check to pass (double submit method).

Yep you're right, we do require a double confirmation. Going to see what the other maintainers are thinking.

**Comment by hi019** [comment]
Whoops, it's accessible via c.Locals. https://github.com/gofiber/fiber/tree/master/middleware/csrf#config - see ContextKey. `c.Locals(contextKey)`

Will add to docs
